### ISSUE TITLE
configurable image paths to delete

### ISSUE THREAD

**Issue by glensc** [glensc/gitlab-registry-cleanup-hook#2]
the original gist by @morph027 deleted `project_path:branch_name`,
my projects use `project_path/branches:project_path:branch_name`

but this should be configurable

also, i have projects that build more than one image per branch:
- `project_path/branches:project_path-build1`
- `project_path/branches:project_path-build2`

not really sure how to make this configurable, especially the second scenario without making the hook code too complex.

**Comment by glensc** [comment]
looks like there's api to set/fetch custom attributes for project:
- https://docs.gitlab.com/ce/api/custom_attributes.html

could utilize that for per-project customizations

**Comment by glensc** [comment]
i'm thinking such config

```sh

# template to use to construct image to cleanup
IMAGE_NAME_TEMPLATE=%(project_path)s/branches:%(branch)s

# Project custom attribute name to set per project override for IMAGE_NAME_TEMPLATE
# https://docs.gitlab.com/ce/api/custom_attributes.html
IMAGE_NAME_PROJECT_ATTRIBUTE=gitlab_registry_image_template
```

and later:

```py
    attributes = {
        'branch': data['object_attributes']['source_branch'],
        'project_path': data['object_attributes']['source']['path_with_namespace'],
    }

    for template in image_template.split(','):
        image = template % attributes
        yield image
```

**Comment by glensc** [comment]
released as 0.5.0
